sdk/storage/azure-storage-blob/azure/storage/blob/container_client.py

```diff
@@ -744,6 +752,7 @@ def walk_blobs(
             results_per_page=results_per_page,
             delimiter=delimiter)
 
+    @distributed_trace
```

Comment:
@rakshith91 Btw is the fix for the SAS URL encoding bug raised earlier in this PR?

Reply:
@zezha-msft No, this is for tracing network calls which has been implemented across different SDKs by @SuyogSoti

Reply:
Yah, I left the comment here because this method `upload_blob` calls `self.get_blob_client` which has the bug. 

Do you know if the fix is done somewhere else? Should I submit a PR for it?

Reply:
@zezha-msft - yes this PR has the fix. However it was not just with `upload_blob`, nor `get_blob_client` - the issue was with _**all**_ SAS URLs.
What I didn't realize is that `parse_qs` function used [here](https://github.com/Azure/azure-sdk-for-python/blob/storage-preview2/sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py#L273) was actually URL-decoding the query string. So in this PR we now use `quote()` to re-encode the SAS token on [the next line](https://github.com/Azure/azure-sdk-for-python/blob/storage-preview2/sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py#L274).

Reply:
Thanks @annatisch, I'll try to finish looking through the PR as soon as possible.